source/elements/oneMKL/source/domains/data_fitting/splines.rst

```diff
@@ -0,0 +1,177 @@
+        FpType* input_data,
+        function_hint FunctionHint = storage_hint::row_major);
+
+      spline& set_coefficients(
+        FpType* data,
+        coefficient_hint CoeffHint = storage_hint::row_major);
+
+      bool is_initialized() const;
+
+      std::int64_t get_required_coeffs_size() const;
+
+      sycl::event construct(const std::vector<sycl::event>& dependencies = {});
+  };
+
+An instance of ``spline<T, ST, N>`` create the ``N``-dimensional spline
+that operates with the ``T`` data type. ``ST`` is a type of spline.
+``T`` can only be ``float`` or ``double``.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Constructor
+     - Description
+   * - ``spline(const sycl::queue& q, std::int64_t ny = 1);``
+     - Create an object with the ``q`` SYCL queue and ``ny`` number of functions.
+   * - ``spline(const sycl::device& dev, const sycl::context& ctx, std::int64_t ny = 1);``
+     - Create an object using the ``dev`` SYCL device, the ``ctx`` context
+       and ``ny`` number of functions.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Member function
+     - Description
+   * - ``spline& set_partitions(FpType* input_data, std::int64_t nx, partition_hint PartitionHint = partition_hint::non_uniform);``
+     - Set partition values that are specified by the ``input_data`` memory pointer and ``nx`` partition values.
+       Users can provide ``PartitionHint`` to specify the layout of data.
+       Default layout is ``non_uniform``.
+       If ``uniform`` is specified, ``nx`` must equals to ``2`` and
+       ``input_data`` must contain only 2 values the left and the right borders of partition.
+       Otherwise, behavior is undefined.
+       If ``input_data`` layout doesn't satisfy ``PartitionHint``, behavior is undefined.
+       Returns a reference to the spline object for which partitions are set.
+
+       Example, for ``uniform``. Let :math:`\left\{ i \right\}_{i=1,\cdots,n}` is a partition.
+       So, ``input_data`` must contain only 2 values: 1, ``n``.
+   * - ``spline& set_function_values(FpType* input_data, function_hint FunctionHint = storage_hint::row_major);``
+     - Set function values that are specified by the ``input_data`` memory pointer.
+       Number of function values must equals to ``ny * nx`` elements.
+       Users can provide ``FunctionHint`` to specify the layout of data.
+       Default layout is ``row_major``.
+       If ``input_data`` layout doesn't satisfy ``FunctionHint``, behavior is undefined.
+       Returns a reference to the spline object for which function values are set.
+   * - ``spline& set_coefficients(FpType* data, coefficient_hint CoeffHint = storage_hint::row_major);``
+     - Set coefficients that are specified by the ``data`` memory pointer.
+       Number of coefficients in the memory must equals to the return value of ``get_required_coeffs_size()``.
+       Users can provide ``CoeffHint`` to specify the layout of data.
+       Default layout is ``row_major``.
```

Comment:
Users can provide a ``CoeffHint`` to specify the layout of data, with the default layout of ``row_major``.